modules/apps/foundation/user-associated-data/user-associated-data-web/src/main/java/com/liferay/user/associated/data/web/model/UADService.java

```diff
@@ -0,0 +1,47 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.user.associated.data.web.model;
+
+import java.util.List;
+
+import org.osgi.service.component.annotations.Component;
+
+/**
+ * @author William Newbury
+ */
+@Component(immediate = true, service = UADService.class)
+public interface UADService {
```

Comment:
We probably want to use a term other than "Service" since in Liferay that term usually refers to a Service Builder generated service.

Reply:
Yeah, we struggled somewhat with the naming when we were planning all this out, never came to anything. When I was prototyping, I just used what I called it in my notes. 

Entity sounds good as a replacement for Asset. What do you think about renaming service from UADService -> UADEntityProcessor or UADEntityManager?

Reply:
UADEntityProcessor is not bad.

What I still miss though is some mention to either exporting or anonymizing the user's data, which is really the whole purpose of this class. In fact, as I write this, I wonder if this class should be broken in two parts: UADEntityAnonymizer and UADEntityExporter. That way it the name makes much more clear the purpose and it also allows implementing one without the other.

Thoughs?

Reply:
Will, Drew and I discussed this and concluded that UADEntityAnonymizer and UADEntityExporter were good concepts. Also, we want to add UADEntityAggregator such that UADEntityAnonymizer and UADEntityExporter can both utilize it for retrieval and the data can be retrieved for viewing independently of exporting it or anonymizing it.